core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -454,6 +454,28 @@ public Table emptyCopy(int rowSize) {
         tables[1] = where(table2Selection);
         return tables;
     }
+    
+    /**
+     * Splits the table into two stratified samples, this uses the specified column to divide the table into groups, randomly assigning records to each according to the proportion given in trainingProportion. 
+     * @param column the column to be used for the stratified sampling
+     * @param table1Proportion The proportion to go in the first table
+     * @return An array two tables, with the first table having the proportion specified in the method parameter,
+     * and the second table having the balance of the rows
+     */
```

Comment:
IIRC the tables will be returned based on the alphabetical order of the category column. That should probably be specified in the comments because the proportion variable is associated with the first table. 

Should the method name indicate that it's only for two-way splits?  It seems like it might be useful in the future to support n-way splits where n is the cardinality of the category column.  If we ever implement the general solution, it might be helpful to name this method so it's clear that it isn't general.

Reply:
If we made it more general, we'd have to change the parameters, so I tend to be okay with the name as it is

Reply:
The more I look at this, the less I think it should be added as written. 

It's not really like the existing split method. which isn't dependent on the contents of a column. It just randomly divides the table into two parts based on the specified proportion.  The use case is creating a simple hold-out sample for testing a model. The order of the tables returned is clear, the first table returned is the one with the specified size or proportion. 

This one claims to be doing stratified sampling. According to wikipedia: 

> "The strata should define a partition of the population. That is, it should be collectively exhaustive and mutually exclusive: every element in the population must be assigned to one and only one stratum."

Based on that definition (which I believe to be the common one), this method is problematic. What happens if there are more than two categories in the input column? You get a result, but any data in categories 3 through n is simply lost. That violates the rule that every element be assigned to a stratum and is a bug waiting to happen. 

The second issue is that the result is dependent on the sorting of the tables returned, which I expect will be alphabetical by category name, although the split method doesn't specify that explicitly so conceivably that could change. Even if it doesn't, this is another bug waiting to happen. You could use it in an application and the categories are renamed, or another category could be added with a name that sorts first.

Let me go back to wikipedia: 

> Stratified sampling strategies
Proportionate allocation uses a sampling fraction in each of the strata that is proportional to that of the total population. For instance, if the population consists of X total individuals, m of which are male and f female (and where m + f = X), then the relative size of the two samples (x1 = m/X males, x2 = f/X females) should reflect this proportion.
Optimum allocation (or disproportionate allocation) - The sampling fraction of each stratum is proportionate to both the proportion (as above) and the standard deviation of the distribution of the variable. Larger samples are taken in the strata with the greatest variability to generate the least possible overall sampling variance.

These are the only two strategies wikipedia lists. The first one, I think, is the usual approach. That's what this method should do by default, and I would love to have a method that does that, or better still, two methods that implement both approaches. 

This method is arbitrary about what goes in each strata. If you want arbitrarily sized samples by category and only want to handle the case where there are two categories, I think that's a very specific use case that should be left to the developer to handle. 

This should be re-written to implement proportional sampling for n categories or it should be removed.

Reply:
> What happens if there are more than two categories in the input column? You get a result, but any data in categories 3 through n is simply lost.

I don't think so. Categories 3 and onward will be sampled according to the given proportion and then added to the result arrays. I do think the code is slightly confusingly written and also read it the same way you did at first, but I don't think what you described is what's actually happening

> This one claims to be doing stratified sampling.

I may have misunderstood the Wikipedia article or the code, but I think this does do proportionate allocation